lib/rbac/filterer.rb

```diff
@@ -252,12 +257,36 @@ def search(options = {})
               .includes(include_for_find).includes(exp_includes)
               .order(order)
 
-      scope = include_references(scope, klass, include_for_find, exp_includes, skip_references)
       scope = scope.limit(limit).offset(offset) if attrs[:apply_limit_in_sql]
-      targets = scope
 
-      unless options[:skip_counts]
-        auth_count = attrs[:apply_limit_in_sql] && limit ? targets.except(:offset, :limit, :order).count(:all) : targets.length
+      # putting this into an inner view will allow us to apply the virtual attributes to LIMIT rows not all of them.
+      if !options[:use_sql_view]
+        scope = include_references(scope, klass, include_for_find, exp_includes, skip_references)
+        targets = scope
+
+        unless options[:skip_counts]
+          auth_count = attrs[:apply_limit_in_sql] && limit ? targets.except(:offset, :limit, :order).count(:all) : targets.length
+        end
+      else
+        if !options[:skip_counts] && (attrs[:apply_limit_in_sql] && limit)
+          auth_count = scope.except(:offset, :limit, :order).count(:all)
+        end
+        if attrs[:apply_limit_in_sql] && limit && extra_cols
+          scope.includes_values = []
+          scope.references_values = []
+          scope = add_joins(klass, scope, include_for_find)
+          scope = add_joins(klass, scope, exp_includes)
+          scope = scope.select(klass.arel_table[Arel.star]) unless scope.select_values&.include?(/\*/)
```

Comment:
> scope.select_values&.include?(/\*/)

Pedantic:  Still hate the look of the "safe navigation operator"... just saying.  (aka:  don't change, `</rant>`)

Reply:
it is growing on me :(
(I know it is anti-growing on you)